plan/logical_plan_builder.go

```diff
@@ -1456,6 +1490,11 @@ func (b *planBuilder) buildSelect(sel *ast.SelectStmt) LogicalPlan {
 			defer b.popTableHints()
 		}
 	}
+	if sel.SelectStmtOpts != nil {
+		origin := b.inStraightJoin
```

Comment:
How about there is a subquery in this select statement?

Reply:
subquery will be rewritten into semi join or join with on conditions,  these kind of joins will not be reordered. So maybe it's ok that `Straight_Join` has no effect on subquery ?

Reply:
So what about query like `select STRAIGHT_JOIN ... from ..., (select ... from t1 join t2 join t3...)`?
Should the inner select be affected by the hint?